src/pages/api/download-reports/abstracts.ts

```diff
@@ -0,0 +1,84 @@
+import { NextApiRequest, NextApiResponse } from "next"
+import { gql } from '@apollo/client';
+import apolloClient from '../../../utils/apolloClient'
+import { WorkType } from "../../doi.org/[...doi]";
+
+const QUERY = gql`
+  query getOrganizationQuery(
+    $id: ID
+    $gridId: ID
+    $crossrefFunderId: ID
+    $cursor: String
+    $filterQuery: String
+    $published: String
+    $resourceTypeId: String
+    $fieldOfScience: String
+    $language: String
+    $license: String
+    $registrationAgency: String
+  ) {
+    organization(
+      id: $id
+      gridId: $gridId
+      crossrefFunderId: $crossrefFunderId
+    ) {
+      works(
+        first: 200
+        after: $cursor
+        query: $filterQuery
+        published: $published
+        resourceTypeId: $resourceTypeId
+        fieldOfScience: $fieldOfScience
+        language: $language
+        license: $license
+        registrationAgency: $registrationAgency
+      ) {
+        nodes {
+          ...WorkFragment
+        }
+      }
+    }
+  }
+  fragment WorkFragment on Work {
+    titles {
+      title
+    }
+    descriptions {
+      description
+      descriptionType
+    }
+    doi
+    publicationYear
+  }
+`
+
+type WorkInfo = Pick<WorkType, 'titles' | 'descriptions' | 'doi' | 'publicationYear'>
+
+const SEP = '\t'
+function formatLine(w: WorkInfo) {
+  return `${w.titles[0] ? w.titles[0].title : ''}${SEP}${w.publicationYear}${SEP}${w.doi}${SEP}${w.descriptions[0] ? w.descriptions[0].description : ''}`
+}
+
+function formatResults(data: WorkInfo[]) {
```

Comment:
Doing the formatting by hand here seems like a place we might make some unintentional mistakes.
I'd recomment pulling in an npm library for csv formatting.  For example fast-csv
https://c2fo.github.io/fast-csv/

Reply:
I replaced all the csv formatting using [csv-stringify](https://www.npmjs.com/package/csv-stringify)